CommunicationLayer/CommDeviceControl/UdpMessageForwarder.cpp

```diff
@@ -26,15 +26,19 @@
 #include <QByteArray>
 #include <QDebug>
 #include <QHostAddress>
+#include <QTextCodec>
+
 
 #include "InfrastructureLayer/Settings/I_Settings.h"
 #include "UdpMessageForwarder.h"
 
+
 UdpMessageForwarder::UdpMessageForwarder(I_Settings& settings)
-: ipAddress_(settings.ipAddress())
-, port_(settings.udpPort())
 {
-    socket_.bind(ipAddress_, port_);
+    routingKey_ = settings.routingKey();
+    exchangeName_ = settings.exchangeName();
+    channel_ = Channel::Create(settings.ipAddress().toStdString(), (int)settings.udpPort());
```

Comment:
Let's keep the same type for `settings.updPort()` so it's not necessary to cast it to a different type.